tests/test_pyrdl.py

```diff
@@ -129,6 +145,32 @@ def test_any_attr_verify():
     any_constraint.verify(IntData(0), ConstraintContext())
 
 
+def test_anyof_verify():
+    """
+    Check that an AnyOf constraint verifies if one of the constraints
+    verify.
+    """
+    constraint = BaseAttr(BoolData) | BaseAttr(IntData)
+    constraint.verify(IntData(1), ConstraintContext())
+    constraint.verify(IntData(-10), ConstraintContext())
+    constraint.verify(BoolData(True), ConstraintContext())
+    constraint.verify(BoolData(False), ConstraintContext())
+
+
+def test_anyof_verify_fail():
+    """
+    Check that an AnyOf constraint fails to verify if none of the constraints
+    verify.
+    """
+    constraint = BaseAttr(BoolData) | BaseAttr(IntData)
+
+    f = FloatData(10.0)
+
+    with pytest.raises(VerifyException) as e:
+        constraint.verify(f, ConstraintContext())
+    assert e.value.args[0] == f"Unexpected attribute {f}"
```

Comment:
Not a fan of format strings in match cases
```suggestion
    with pytest.raises(VerifyException, match="Unexpected attribute 10.0"):
        constraint.verify(f, ConstraintContext())
```